compiler/transforms/accfg_config_overlap.py

```diff
@@ -53,10 +57,30 @@ def match_and_rewrite(self, op: accfg.SetupOp, rewriter: PatternRewriter, /):
         if launch.next_op is op:
             return
 
-        # detach the setup op
-        op.detach()
-        # insert the op right after the launch
-        rewriter.insert_op_after(op, launch)
+        # grab the parent block, which can't be none (we know that)
+        parent_block = op.parent_block()
+        assert parent_block is not None
+
+        # grab the setup op with all inputs in this block
+        inputs = get_scoped_setup_inputs(op, parent_block)
+
+        # if we have immovable inputs, abort
+        # TODO: it is fine if the immovable inputs are *before* the launch op anyway, as we only
+        #       care about ops up until the launch op. We can implement this later if it's a problem.
+        if inputs is None:
+            return
+
+        # if all the ops between launch and setup are input ops, then there's nothing to move!
+        if all(
+            between_op in inputs.inputs for between_op in iter_ops_range(launch, op)
+        ):
+            return
+
+        inputs.lazy_move_up(  # and move the setup and inputs to be right behind the launch
+            parent_block,
+            InsertPoint.after(launch),
+            rewriter,
+        )
```

Comment:
Maybe we should make clear that this is a heuristic? We can move these ops anywhere in the block iiuc, and not all possible move strategies might make as much sense depending on the scenario?

Reply:
Not sure what you mean exactly by this...